### ISSUE TITLE
Webapp:Set-AzureWebsite - 410 Response during site update

### ISSUE THREAD

**Issue by carlmwright** [Azure/azure-powershell#7558]
### Description

We are trying to use existing scripts to update AppSettings and ConnectionStrings on production and staging slots of our application.

We are getting the following error message:

Set-AzureWebsite : <string xmlns="http://schemas.microsoft.com/2003/10/Serialization/">Request to a downlevel service
failed.</string>
At F:\Scripts\xxx-Production-settings.ps1:199 char:1
+ Set-AzureWebsite -SiteWithConfig $site  -Name $websiteName -Slot $web ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Set-AzureWebsite], CloudException
    + FullyQualifiedErrorId : Microsoft.WindowsAzure.Commands.Websites.SetAzureWebsiteCommand

### Script/Steps for Reproduction

```powershell
$site = Get-AzureWebsite -Name appname -Slot production

Set-AzureWebsite -Name appname -Slot production -SiteWithConfig $site
```

### Module Version

```powershell
ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Script     5.3.0      Azure                               {Get-AzureAutomationCertificate, Get-AzureAutomationConnec...
Script     0.5.4      Azure.AnalysisServices              {Add-AzureAnalysisServicesAccount, Restart-AzureAnalysisSe...
Script     0.5.1      Azure.AnalysisServices              {Add-AzureAnalysisServicesAccount, Restart-AzureAnalysisSe...
Script     4.6.1      Azure.Storage                       {Get-AzureStorageTable, New-AzureStorageTableSASToken, New...
Script     4.3.0      Azure.Storage                       {Get-AzureStorageTable, New-AzureStorageTableSASToken, New...
Script     6.10.0     AzureRM
Script     0.6.14     AzureRM.AnalysisServices            {Resume-AzureRmAnalysisServicesServer, Suspend-AzureRmAnal...
Script     0.6.8      AzureRM.AnalysisServices            {Resume-AzureRmAnalysisServicesServer, Suspend-AzureRmAnal...
Script     6.1.5      AzureRM.ApiManagement               {Add-AzureRmApiManagementRegion, Get-AzureRmApiManagementS...
Script     6.1.0      AzureRM.ApiManagement               {Add-AzureRmApiManagementRegion, Get-AzureRmApiManagementS...
Script     0.1.8      AzureRM.ApplicationInsights         {Get-AzureRmApplicationInsights, New-AzureRmApplicationIns...
Script     0.1.4      AzureRM.ApplicationInsights         {Get-AzureRmApplicationInsights, New-AzureRmApplicationIns...
Script     5.1.1      AzureRM.Automation                  {Get-AzureRMAutomationHybridWorkerGroup, Get-AzureRmAutoma...
Script     5.0.0      AzureRM.Automation                  {Get-AzureRMAutomationHybridWorkerGroup, Get-AzureRmAutoma...
Script     4.0.10     AzureRM.Backup                      {Backup-AzureRmBackupItem, Enable-AzureRmBackupContainerRe...
Script     4.0.5      AzureRM.Backup                      {Backup-AzureRmBackupItem, Enable-AzureRmBackupContainerRe...
Script     4.1.5      AzureRM.Batch                       {Remove-AzureRmBatchAccount, Get-AzureRmBatchAccount, Get-...
Script     4.1.0      AzureRM.Batch                       {Remove-AzureRmBatchAccount, Get-AzureRmBatchAccount, Get-...
Script     0.14.6     AzureRM.Billing                     {Get-AzureRmBillingInvoice, Get-AzureRmBillingPeriod, Get-...
Script     0.14.2     AzureRM.Billing                     {Get-AzureRmBillingInvoice, Get-AzureRmBillingPeriod, Get-...
Script     5.0.6      AzureRM.Cdn                         {Get-AzureRmCdnProfile, Get-AzureRmCdnProfileSsoUrl, New-A...
Script     5.0.0      AzureRM.Cdn                         {Get-AzureRmCdnProfile, Get-AzureRmCdnProfileSsoUrl, New-A...
Script     0.9.11     AzureRM.CognitiveServices           {Get-AzureRmCognitiveServicesAccount, Get-AzureRmCognitive...
Script     0.9.5      AzureRM.CognitiveServices           {Get-AzureRmCognitiveServicesAccount, Get-AzureRmCognitive...
Script     5.7.0      AzureRM.Compute                     {Remove-AzureRmAvailabilitySet, Get-AzureRmAvailabilitySet...
Script     5.0.0      AzureRM.Compute                     {Remove-AzureRmAvailabilitySet, Get-AzureRmAvailabilitySet...
Script     0.3.7      AzureRM.Consumption                 {Get-AzureRmConsumptionBudget, Get-AzureRmConsumptionMarke...
Script     0.3.3      AzureRM.Consumption                 Get-AzureRmConsumptionUsageDetail
Script     0.2.10     AzureRM.ContainerInstance           {New-AzureRmContainerGroup, Get-AzureRmContainerGroup, Rem...
Script     0.2.6      AzureRM.ContainerInstance           {New-AzureRmContainerGroup, Get-AzureRmContainerGroup, Rem...
Script     1.0.10     AzureRM.ContainerRegistry           {New-AzureRmContainerRegistry, Get-AzureRmContainerRegistr...
Script     1.0.5      AzureRM.ContainerRegistry           {New-AzureRmContainerRegistry, Get-AzureRmContainerRegistr...
Script     5.0.3      AzureRM.DataFactories               {Remove-AzureRmDataFactory, Get-AzureRmDataFactoryRun, Get...
Script     5.0.0      AzureRM.DataFactories               {Remove-AzureRmDataFactory, Get-AzureRmDataFactoryRun, Get...
Script     0.5.11     AzureRM.DataFactoryV2               {Set-AzureRmDataFactoryV2, Update-AzureRmDataFactoryV2, Ge...
Script     0.5.4      AzureRM.DataFactoryV2               {Set-AzureRmDataFactoryV2, Update-AzureRmDataFactoryV2, Ge...
Script     5.1.4      AzureRM.DataLakeAnalytics           {Get-AzureRmDataLakeAnalyticsDataSource, New-AzureRmDataLa...
Script     5.0.0      AzureRM.DataLakeAnalytics           {Get-AzureRmDataLakeAnalyticsDataSource, New-AzureRmDataLa...
Script     6.1.2      AzureRM.DataLakeStore               {Get-AzureRmDataLakeStoreTrustedIdProvider, Remove-AzureRm...
Script     6.0.1      AzureRM.DataLakeStore               {Get-AzureRmDataLakeStoreTrustedIdProvider, Remove-AzureRm...
Script     4.0.9      AzureRM.DevTestLabs                 {Get-AzureRmDtlAllowedVMSizesPolicy, Get-AzureRmDtlAutoShu...
Script     4.0.5      AzureRM.DevTestLabs                 {Get-AzureRmDtlAllowedVMSizesPolicy, Get-AzureRmDtlAutoShu...
Script     5.1.0      AzureRM.Dns                         {Get-AzureRmDnsRecordSet, New-AzureRmDnsRecordConfig, Remo...
Script     5.0.0      AzureRM.Dns                         {Get-AzureRmDnsRecordSet, New-AzureRmDnsRecordConfig, Remo...
Script     0.3.7      AzureRM.EventGrid                   {New-AzureRmEventGridTopic, Get-AzureRmEventGridTopic, Set...
Script     0.3.3      AzureRM.EventGrid                   {New-AzureRmEventGridTopic, Get-AzureRmEventGridTopic, Set...
Script     0.6.10     AzureRM.EventHub                    {New-AzureRmEventHubNamespace, Get-AzureRmEventHubNamespac...
Script     0.6.4      AzureRM.EventHub                    {New-AzureRmEventHubNamespace, Get-AzureRmEventHubNamespac...
Script     4.1.8      AzureRM.HDInsight                   {Get-AzureRmHDInsightJob, New-AzureRmHDInsightSqoopJobDefi...
Script     4.1.3      AzureRM.HDInsight                   {Get-AzureRmHDInsightJob, New-AzureRmHDInsightSqoopJobDefi...
Script     5.1.4      AzureRM.Insights                    {Get-AzureRmMetricDefinition, Get-AzureRmMetric, Remove-Az...
Script     5.0.0      AzureRM.Insights                    {Get-AzureRmMetricDefinition, Get-AzureRmMetric, Remove-Az...
Script     3.1.8      AzureRM.IotHub                      {Add-AzureRmIotHubKey, Get-AzureRmIotHubEventHubConsumerGr...
Script     3.1.3      AzureRM.IotHub                      {Add-AzureRmIotHubKey, Get-AzureRmIotHubEventHubConsumerGr...
Script     5.2.1      AzureRM.KeyVault                    {Add-AzureKeyVaultCertificate, Update-AzureKeyVaultCertifi...
Script     5.0.0      AzureRM.KeyVault                    {Add-AzureKeyVaultCertificate, Update-AzureKeyVaultCertifi...
Script     4.1.4      AzureRM.LogicApp                    {Get-AzureRmIntegrationAccountAgreement, Get-AzureRmIntegr...
Script     4.0.4      AzureRM.LogicApp                    {Get-AzureRmIntegrationAccountAgreement, Get-AzureRmIntegr...
Script     0.18.5     AzureRM.MachineLearning             {Move-AzureRmMlCommitmentAssociation, Get-AzureRmMlCommitm...
Script     0.18.0     AzureRM.MachineLearning             {Move-AzureRmMlCommitmentAssociation, Get-AzureRmMlCommitm...
Script     0.4.8      AzureRM.MachineLearningCompute      {Get-AzureRmMlOpCluster, Get-AzureRmMlOpClusterKey, Test-A...
Script     0.4.3      AzureRM.MachineLearningCompute      {Get-AzureRmMlOpCluster, Get-AzureRmMlOpClusterKey, Test-A...
Script     0.2.5      AzureRM.MarketplaceOrdering         {Get-AzureRmMarketplaceTerms, Set-AzureRmMarketplaceTerms}
Script     0.2.2      AzureRM.MarketplaceOrdering         {Get-AzureRmMarketplaceTerms, Set-AzureRmMarketplaceTerms}
Script     0.10.4     AzureRM.Media                       {Sync-AzureRmMediaServiceStorageKeys, Set-AzureRmMediaServ...
Script     0.10.0     AzureRM.Media                       {Sync-AzureRmMediaServiceStorageKeys, Set-AzureRmMediaServ...
Script     6.9.0      AzureRM.Network                     {Add-AzureRmApplicationGatewayAuthenticationCertificate, G...
Script     6.1.0      AzureRM.Network                     {Add-AzureRmApplicationGatewayAuthenticationCertificate, G...
Script     5.0.3      AzureRM.NotificationHubs            {Get-AzureRmNotificationHub, Get-AzureRmNotificationHubAut...
Script     5.0.0      AzureRM.NotificationHubs            {Get-AzureRmNotificationHub, Get-AzureRmNotificationHubAut...
Script     5.0.6      AzureRM.OperationalInsights         {New-AzureRmOperationalInsightsAzureActivityLogDataSource,...
Script     5.0.0      AzureRM.OperationalInsights         {New-AzureRmOperationalInsightsAzureActivityLogDataSource,...
Script     1.0.4      AzureRM.PolicyInsights              {Get-AzureRmPolicyEvent, Get-AzureRmPolicyState, Get-Azure...
Script     4.1.10     AzureRM.PowerBIEmbedded             {Remove-AzureRmPowerBIWorkspaceCollection, Get-AzureRmPowe...
Script     4.1.5      AzureRM.PowerBIEmbedded             {Remove-AzureRmPowerBIWorkspaceCollection, Get-AzureRmPowe...
Script     5.6.0      AzureRM.Profile                     {Disable-AzureRmDataCollection, Disable-AzureRmContextAuto...
Script     5.1.0      AzureRM.Profile                     {Disable-AzureRmDataCollection, Disable-AzureRmContextAuto...
Script     4.1.8      AzureRM.RecoveryServices            {Get-AzureRmRecoveryServicesBackupProperty, Get-AzureRmRec...
Script     4.1.3      AzureRM.RecoveryServices            {Get-AzureRmRecoveryServicesBackupProperty, Get-AzureRmRec...
Script     4.4.1      AzureRM.RecoveryServices.Backup     {Backup-AzureRmRecoveryServicesBackupItem, Get-AzureRmReco...
Script     4.1.3      AzureRM.RecoveryServices.Backup     {Backup-AzureRmRecoveryServicesBackupItem, Get-AzureRmReco...
Script     0.2.10     AzureRM.RecoveryServices.SiteRec... {Edit-AzureRmRecoveryServicesAsrRecoveryPlan, Get-AzureRmR...
Script     0.2.5      AzureRM.RecoveryServices.SiteRec... {Edit-AzureRmRecoveryServicesAsrRecoveryPlan, Get-AzureRmR...
Script     5.1.0      AzureRM.RedisCache                  {Remove-AzureRmRedisCachePatchSchedule, New-AzureRmRedisCa...
Script     5.0.0      AzureRM.RedisCache                  {Remove-AzureRmRedisCachePatchSchedule, New-AzureRmRedisCa...
Script     0.3.10     AzureRM.Relay                       {New-AzureRmRelayNamespace, Get-AzureRmRelayNamespace, Set...
Script     0.3.4      AzureRM.Relay                       {New-AzureRmRelayNamespace, Get-AzureRmRelayNamespace, Set...
Script     6.6.0      AzureRM.Resources                   {Get-AzureRmProviderOperation, Remove-AzureRmRoleAssignmen...
Script     6.0.0      AzureRM.Resources                   {Get-AzureRmProviderOperation, Remove-AzureRmRoleAssignmen...
Script     0.16.10    AzureRM.Scheduler                   {Disable-AzureRmSchedulerJobCollection, Enable-AzureRmSche...
Script     0.16.4     AzureRM.Scheduler                   {Disable-AzureRmSchedulerJobCollection, Enable-AzureRmSche...
Script     0.6.12     AzureRM.ServiceBus                  {New-AzureRmServiceBusNamespace, Get-AzureRmServiceBusName...
Script     0.6.5      AzureRM.ServiceBus                  {New-AzureRmServiceBusNamespace, Get-AzureRmServiceBusName...
Script     0.3.12     AzureRM.ServiceFabric               {Add-AzureRmServiceFabricApplicationCertificate, Add-Azure...
Script     0.3.6      AzureRM.ServiceFabric               {Add-AzureRmServiceFabricApplicationCertificate, Add-Azure...
Script     1.0.0      AzureRM.SignalR                     {New-AzureRmSignalR, Get-AzureRmSignalR, Get-AzureRmSignal...
Script     4.11.5     AzureRM.Sql                         {Get-AzureRmSqlDatabaseTransparentDataEncryption, Get-Azur...
Script     4.5.0      AzureRM.Sql                         {Get-AzureRmSqlDatabaseTransparentDataEncryption, Get-Azur...
Script     5.2.0      AzureRM.Storage                     {Get-AzureRmStorageAccount, Get-AzureRmStorageAccountKey, ...
Script     5.0.0      AzureRM.Storage                     {Get-AzureRmStorageAccount, Get-AzureRmStorageAccountKey, ...
Script     4.0.10     AzureRM.StreamAnalytics             {Get-AzureRmStreamAnalyticsFunction, Get-AzureRmStreamAnal...
Script     4.0.5      AzureRM.StreamAnalytics             {Get-AzureRmStreamAnalyticsFunction, Get-AzureRmStreamAnal...
Script     4.0.5      AzureRM.Tags                        {Remove-AzureRmTag, Get-AzureRmTag, New-AzureRmTag}
Script     4.0.2      AzureRM.Tags                        {Remove-AzureRmTag, Get-AzureRmTag, New-AzureRmTag}
Script     4.1.1      AzureRM.TrafficManager              {Add-AzureRmTrafficManagerCustomHeaderToEndpoint, Remove-A...
Script     4.0.5      AzureRM.TrafficManager              {Disable-AzureRmTrafficManagerEndpoint, Enable-AzureRmTraf...
Script     4.0.5      AzureRM.UsageAggregates             Get-UsageAggregates
Script     4.0.3      AzureRM.UsageAggregates             Get-UsageAggregates
Script     5.2.0      AzureRM.Websites                    {Get-AzureRmAppServicePlan, Set-AzureRmAppServicePlan, New...
Script     5.0.1      AzureRM.Websites                    {Get-AzureRmAppServicePlan, Set-AzureRmAppServicePlan, New...
Script     1.0.1      Microsoft.PowerShell.Operation.V... {Get-OperationValidation, Invoke-OperationValidation}
Binary     1.0.0.1    PackageManagement                   {Find-Package, Get-Package, Get-PackageProvider, Get-Packa...
Script     3.4.0      Pester                              {Describe, Context, It, Should...}
Script     1.0.0.1    PowerShellGet                       {Install-Module, Find-Module, Save-Module, Update-Module...}
Script     1.2        PSReadline                          {Get-PSReadlineKeyHandler, Set-PSReadlineKeyHandler, Remov...
```

### Environment Data

```powershell
Name                           Value
----                           -----
PSVersion                      5.1.17134.228
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.17134.228
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```
### Debug Output

```
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
POST

Absolute Uri:
https://management.core.windows.net/xxx/services/WebSpaces/xxx-NorthEuropewebspace/sites/.../slotConfigNames

Headers:
accept                        : application/json
x-ms-version                  : 2014-04-01

Body:
{
  "AppSettingNames": []
}

DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
Gone

Headers:
Date                          : Mon, 15 Oct 2018 10:42:39 GMT
Server                        : Microsoft-HTTPAPI/2.0

Body:
<string xmlns="http://schemas.microsoft.com/2003/10/Serialization/">Request to a downlevel service failed.</string>

Set-AzureWebsite : <string xmlns="http://schemas.microsoft.com/2003/10/Serialization/">Request to a downlevel service
failed.</string>
At F:\Scripts\xxx-Production-settings.ps1:199 char:1
+ Set-AzureWebsite -SiteWithConfig $site  -Name $websiteName -Slot $web ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Set-AzureWebsite], CloudException
    + FullyQualifiedErrorId : Microsoft.WindowsAzure.Commands.Websites.SetAzureWebsiteCommand

```

**Comment by markcowl** [comment]
@carlmwright Can you please share the details of how you are setting the ApPSettins and Configuration in SiteConfig?

@panchagnula Can you take a look?

**Comment by carlmwright** [comment]
Hi,

I don't have to set any of them programmatically in order to produce the error. 

I get the same result if I have a site with no AppSettings or ConnectionStrings set, or if they have been set through the portal.

C

**Comment by DorkaitzVe** [comment]
Hi,

Same problem here. We reported it in here --> [https://developercommunity.visualstudio.com/comments/357955/view.html](url)

> @carlmwright Can you please share the details of how you are setting the ApPSettins and Configuration in SiteConfig?

$productionSlot = Get-AzureWebsite -Name #{WebAppName} -Slot "Production"
$appSettingsProduction = $productionSlot.AppSettings
 
$appSettingsProduction.Add("Deploying", "True")
Set-AzureWebsite -Name #{WebAppName} -Slot "Production" -AppSettings $appSettingsProduction

**Comment by KumG** [comment]
Hello,

Same problem here for "New-AzureWebSite" command :

Script :

Import-AzurePublishSettingsFile -PublishSettingsFile "myFile.publishsettings" -Verbose -ErrorAction Stop

Select-AzureSubscription "MySubscription" -Verbose -ErrorAction Stop

$website = New-AzureWebSite -Name "myWebSiteName" -Location "West Europe" -Verbose -ErrorAction Stop

Exception :

New-AzureWebSite : <string xmlns="http://schemas.microsoft.com/2003/10/Serialization/">Request to a downlevel service failed.</string> At C:\test.ps1:6 char:12 + $website = New-AzureWebSite -Name "myWebSiteName" -Location "Wes ... + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ + CategoryInfo : CloseError: (:) [New-AzureWebsite], Exception + FullyQualifiedErrorId : Microsoft.WindowsAzure.Commands.Websites.NewAzureWebsiteCommand

**Comment by v-hedlund** [comment]
Facing the same issue when using Set-AzureWebsite

Any updates on this?

**Comment by vany0114** [comment]
I'm facing the same issue but using `New-AzureWebsite`

**Comment by panchagnula** [comment]
Haven't had a chance to look into this in detail. However, need to clarify the version - what version AzureRm.Websites are you using? From above I see 2 versions being used
Script     5.2.0      AzureRM.Websites                    {Get-AzureRmAppServicePlan, Set-AzureRmAppServicePlan, New...
Script     5.0.1      AzureRM.Websites                    {Get-AzureRmAppServicePlan, Set-AzureRmAppServicePlan,

Can we make sure only a single & the latest version AzureRM,Websites is used?

**Comment by tonjohn** [comment]
My team is hitting the same issue.

Script     4.2.2      AzureRM.Websites

The command appears to complete the intended action (the config gets updated correctly) but the error causes our VSTS release pipeline to stop.

If you look at the debug logs, you can see the exact request that is having issues.

### Failing Request
```
POST https://management.core.windows.net/%subscriptionId%/services/WebSpaces/%resourceGroup%-%region%webspace/sites/%siteName%/slotConfigNames

"StatusCode": 410,
"ReasonPhrase": "Gone",
```

### Error Message
```
<string xmlns="http://schemas.microsoft.com/2003/10/Serialization/">Request to a downlevel service failed.</string>
```
### Stack Trace
```
   at Microsoft.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at Microsoft.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccess(Task task)
   at Microsoft.WindowsAzure.Management.WebSites.WebSiteOperationsExtensions.**UpdateSlotConfigNames**(IWebSiteOperations operations, String webSpaceName, String webSiteName, SlotConfigNamesUpdate parameters)
   at Microsoft.WindowsAzure.Commands.Utilities.Websites.WebsitesClient.UpdateWebsiteConfiguration(String name, SiteConfig newConfiguration)
   at Microsoft.WindowsAzure.Commands.Websites.SetAzureWebsiteCommand.UpdateConfig()
   at Microsoft.WindowsAzure.Commands.Websites.SetAzureWebsiteCommand.ExecuteCmdlet()
   at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord()",
```

**Comment by ruslany** [comment]
Azure App Service team is in the process of deprecating the old Service Management APIs as described here: https://blogs.msdn.microsoft.com/appserviceteam/2018/03/12/deprecating-service-management-apis-support-for-azure-app-services/. 

It looks like even though you have the Azure Resource Manager PowerShell cmdlets installed (Script N.N.N AzureRM.Websites) you are still using the old Service Management cmdlets such as Set-AzureWebSite. Here is the list of all the cmdlets that use deprecated API's:

Get-AzureWebsiteMetric
Publish-AzureWebsiteProject
Update-AzureWebsiteRepository
Switch-AzureWebsiteSlot
Disable-AzureWebsiteApplicationDiagnostic
Enable-AzureWebsiteDebug
Disable-AzureWebsiteDebug
Enable-AzureWebsiteApplicationDiagnostic
Get-AzureWebsite
Get-AzureWebsiteDeployment
Get-AzureWebsiteLocation
Get-AzureWebsiteLog
New-AzureWebsite
Remove-AzureWebsite
Restart-AzureWebsite
Restore-AzureWebsiteDeployment
Save-AzureWebsiteLog
Set-AzureWebsite
Show-AzureWebsite
Start-AzureWebsite
Stop-AzureWebsite
Get-AzureWebsiteJobHistory
Stop-AzureWebsiteJob
Start-AzureWebsiteJob
Remove-AzureWebsiteJob
New-AzureWebsiteJob
Get-AzureWebsiteJob

**Comment by ruslany** [comment]
Azure App Service team is in the process of deprecating Service Management APIs as per the announcement here: https://blogs.msdn.microsoft.com/appserviceteam/2018/03/12/deprecating-service-management-apis-support-for-azure-app-services/. New-AzureWebsite, Set-AzureWebsite, etc are the old cmdlets that call the Service Management API endpoint.

We've temporarily unblocked those APIs but we will have to block them again soon as we are in the process of closing down the Service Management API endpoint for App Service. Please change all your automation scripts to use Azure Resource Manager cmdlets (e.g. **Set-AzureRmWebApp**).

Here is the list of all the deprecated cmdlets:

```
$m = Get-Module -ListAvailable | ? {$_.Name -eq "Azure"}
$m.ExportedCmdlets.Keys | where {$_ -match "website"}
Get-AzureWebsiteMetric
Publish-AzureWebsiteProject
Update-AzureWebsiteRepository
Switch-AzureWebsiteSlot
Disable-AzureWebsiteApplicationDiagnostic
Enable-AzureWebsiteDebug
Disable-AzureWebsiteDebug
Enable-AzureWebsiteApplicationDiagnostic
Get-AzureWebsite
Get-AzureWebsiteDeployment
Get-AzureWebsiteLocation
Get-AzureWebsiteLog
New-AzureWebsite
Remove-AzureWebsite
Restart-AzureWebsite
Restore-AzureWebsiteDeployment
Save-AzureWebsiteLog
Set-AzureWebsite
Show-AzureWebsite
Start-AzureWebsite
Stop-AzureWebsite
Get-AzureWebsiteJobHistory
Stop-AzureWebsiteJob
Start-AzureWebsiteJob
Remove-AzureWebsiteJob
New-AzureWebsiteJob
Get-AzureWebsiteJob
```

**Comment by panchagnula** [comment]
Thanks @ruslany . Filed the following to mark these deprecated https://github.com/Azure/azure-powershell/issues/7668, Please use the following equivalent ARM commands https://docs.microsoft.com/en-us/powershell/module/azurerm.websites/?view=azurermps-6.11.0#app_service

**Comment by eyccli** [comment]
@panchagnula  hi, I have a question since I am facing the same issue.  Is there an equivalent command to"Publish-AzureWebsiteProject" ? I do not see anything that supports that in the new ARM commands.

**Comment by bsweeney** [comment]
@eyccli I find myself in a similar situation. I believe you can use the [azure command line](https://docs.microsoft.com/en-us/cli/azure/?view=azure-cli-latest). The equivalent command would be something along the lines of (depending on your current usage):

```
az webapp deployment source config-zip --resource-group <group_name> --name <app_name> --src <filename>.zip
```
[ref](https://docs.microsoft.com/en-us/cli/azure/webapp/deployment/source?view=azure-cli-latest#az-webapp-deployment-source-config-zip)

**Comment by panchagnula** [comment]
@eyccli , right now there is no equivalent for the Publish-AzureWebsiteProject since that uses WebDeploy, currently the ZipDeployment is supported on azure-cli only, we are working getting this support in Powershell as well. @Nking92 , FYI. Please link the ZipDeployment commandline item to this issue when you get a PR started. Thank you.

**Comment by kpfaulkner** [comment]
We're also hitting this suddenly with OctopusDeploy (multiple versions/instances) and have raised a ticket with Azure to see if the APIs are definitely deprecated.

**Comment by eyccli** [comment]
> @eyccli I find myself in a similar situation. I believe you can use the [azure command line](https://docs.microsoft.com/en-us/cli/azure/?view=azure-cli-latest). The equivalent command would be something along the lines of (depending on your current usage):
> 
> ```
> az webapp deployment source config-zip --resource-group <group_name> --name <app_name> --src <filename>.zip
> ```
> [ref](https://docs.microsoft.com/en-us/cli/azure/webapp/deployment/source?view=azure-cli-latest#az-webapp-deployment-source-config-zip)

@bsweeney  Thanks for this, I will give it a try.  Our deployment was very dependent on  using "publishsettings" and the "Get-AzureWebsite", "Set-AzureWebsite" and "Publish-AzureWebsiteProject" from our Bamboo CI. We automated everything without any user interaction, so this is almost a step backwards until an equivalent is in place in AzureRM.  I'm not sure why it was deprecated and removed before a new cmdlet was in place to replace it.

@panchagnula I have found that I can use ZipDeployUI in Kudu for now, since it requires manual interaction, we can just log into Azure and drop the compiled zip file.

**Comment by panchagnula** [comment]
@kpfaulkner , we did unblocked the "Publish-AzureWebsiteProject" APIs for now, until we get the ZipDeployment command in Powershell. We are targeting Mid Jan release for this at this point, this should unblock you for now. Please re-try & confirm that this is infact the case. Thank you!

**Comment by eyccli** [comment]
Can we also unblock the "Set-AzureWebsite"? thank you so much!

**Comment by panchagnula** [comment]
@eyccli Set-AzureWebsite does have an equivalent command for now, & for the missing TIP properties we do have recommended powershell workaround, so those will remain blocked for now. Thank you.

**Comment by kpfaulkner** [comment]
@panchagnula Success! Ok, so I take it this means come mid Jan we're back to a removed API? We'll migrate to Octopus using Service Principals which hopefully will avoid this in the future. Thanks for reverting the change!!

**Comment by bsweeney** [comment]
@eyccli FYI we are using similar setup with publishsettings and Publish-AzureWebsiteProject (but not the other two commands). It looks like you can import publishsettings using the `az account` command. I've not tested yet because I'm waiting for azure cmd to be updated our deployment server.

Good to hear about the temporary unblock for publish, and that equivalent functionality is in development. Thanks for the info, panchagnula.

**Comment by eyccli** [comment]
@panchagnula  It appears that "Publish-AzureWebsiteProject" is working for us this morning.  I'll get my infrastructure team to look into Service Principals and wait for the ZipDeployment PS command.

Regards,

**Comment by chamikasandamal** [comment]
Using TFS 2015 release pipeline with Azure web app deployment. and with this change that is not working.

**Comment by tonjohn** [comment]
Publish-AzureWebsiteProject has been hitting the 'downlevel service failed' error since Friday. Why does this issue keep randomly cropping up?

**Comment by eyccli** [comment]
> Publish-AzureWebsiteProject has been hitting the 'downlevel service failed' error since Friday. Why does this issue keep randomly cropping up?

I bet it's been disabled again like what they did in December. They're supposed to be rolling out AzureRM to replace Azure Powershell functions but last I recall they didn't have a "publish-azurewebsiteproject" equivalent in place.

**Comment by ruslany** [comment]
@tonjohn @eyccli - There is a new cmdlet - Publish-AzWebApp that is a replacement for the Publish-AzureWebSiteProject. (https://github.com/Azure/azure-powershell/pull/8250). Since the new cmdlet is available we've resumed blocking of the deprecated APIs.

**Comment by eyccli** [comment]
@ruslany   I just read Ticket 8250 above and seems to be the case.  Are there any documentation on how to get this to work with a CI so users do not need to interact with deployment, similar to the old Azure cmdlets? Or do we have to use a service principle?  Thanks.

**Comment by panchagnula** [comment]
@Nking92 can you please comment on the above query.  @eyccli this is the only documentation I am aware of https://docs.microsoft.com/en-us/powershell/module/Az.Websites/Publish-AzWebApp?view=azps-1.5.0

**Comment by tonjohn** [comment]
@ruslany why do the errors in the deprecated APIs not make this clear? Instead of a confusing error, they should point me to the new API...

**Comment by jixer** [comment]
I believe this error may go away for most of us as reported by polischuk here: https://github.com/Azure/azure-powershell/issues/8826

Not sure if this will crop up again at some point, but the "request to a downlevel service failed" error when I call Get-AzureWebSite went away today on my side and I made no changes to the script we're using.

**Comment by ruslany** [comment]
@jixer - even though those cmdlets work temporarily I strongly recommend to move away from using them as soon as possible.

**Comment by chaami** [comment]
@ruslany I am in the process of migrating deployment scripts to the latest [Az PowerShell kit](https://docs.microsoft.com/en-us/powershell/module/az.websites/publish-azwebapp?view=azps-1.6.0) as expected.
I now have the [Publish-AzWebApp](https://docs.microsoft.com/en-us/powershell/module/az.websites/publish-azwebapp?view=azps-1.6.0) and everything I need to push App Services.
However, I fail to find an alternate route for deploying Cloud Services.
I understand that it's a legacy offering and that because of that no development regarding it is planned in Az, however, I hope that you will not cut the deployment API (Set-AzureDeployment and what is needed to have the right credentials to push) until they are fully decommissioned from the cloud offering.
In short, I sure hope that the filtering that you did on the exported commandlets in your comment (way) above wasn't just because of the title of the Issue, but because the rest is there to say...
Could you please confirm by any chance ?

**Comment by panchagnula** [comment]
@chaami Is this the command you are referring to https://docs.microsoft.com/en-us/powershell/module/servicemanagement/azure/set-azuredeployment?view=azuresmps-4.0.0

can you tell us exactly what is that in this command that is required for your deployment purposes? The Publish-AzWebApp uses ZipDeployment for the deployment purposes. Thank you.

**Comment by chaami** [comment]
Hi @panchagnula,
Sorry I didn't make myself clear. I am searching for an equivalent command of (Set-AzureDeployment) that is used for Cloud Services deployment (as opposed to App Services).

**Comment by panchagnula** [comment]
@chaami got it. I am aware of the cloud services commandlets, @cormacpayne is there an equivalent AZ command for Set-AzureDeployment???

**Comment by chaami** [comment]
Hi,
Publish-AzureWebsiteProject is down again as of writing...
Is it a temporary glitch or is the final unplugging ?
Also, the Az blob storage commands do not see the classic blob storage, is there a documented solution on how to access the two kind of resources simultaneously given that the Az module and the legacy Azure Module cannot be installed in the same machine ?
Thanks in advance and kind regards.

**Comment by panchagnula** [comment]
@chaami regarding the storage commands please file a new issue since we don't own that command so can't speak for it.

Regarding, Publish-AzureWebsiteProject not aware of any new changes we made here @ruslany can confirm if we did some final unplugging here.

**Comment by chaami** [comment]
Thanks a lot @panchagnula for your quick response.
Right now all my Publish-AzureWebSiteProject calls are failing with a website was not found error.
I can see the website just fine in the Azure portal and nothing has changed in the deployment script...
Kind regards.

**Comment by ruslany** [comment]
@chaami - This is a final unplugging. We are making changes to our infrastructure by regionalizing the Microsoft.Web RP API endpoint so that each region has its own endpoint serving just that region. There are many benefits that this new model provides and we are in the final stage of completing this work. The old APIs that were used by the old powershell cmdlets were not regionalized and that's why they do not work in this new architecture. 

As for the lack of the equivalent for the Set-AzureDeployment cmdlet - have you tried using the generic ARM cmdlets [Set-AzResource](https://docs.microsoft.com/en-us/powershell/module/az.resources/set-azresource?view=azps-2.8.0) and Get-AzResource? With those you can make ARM API calls by specifying the ARM resource id of the cloud service.

**Comment by chaami** [comment]
@ruslany Thanks for pointing me to the Az.Resource module,
I am looking into it to try to find a deployment solution for Cloud Services.
I regret not receiving this information in April and having to rush now while the service is cut.
I don't see any information about it in the [Migrating from AzureRM documentation](https://docs.microsoft.com/en-us/powershell/azure/migrate-from-azurerm-to-az?view=azps-2.8.0)....
Did I miss something ?
Thanks again and kind regards.

**Comment by panchagnula** [comment]
@chaami I recommend you filing a new issue on your question on CloudServices & some from the core PowerShell team or cloudShell might help you. Given this issue is closed & tagged as AppService, highly unlikely other teams will respond to this.

**Comment by chaami** [comment]
For the people that may be struggling with this, please note that `Publish-AzWebApp` will silently fail if you try to publish zip files generated by a Package publish build command (which worked fine with the `Publish-AzureWebsite`).
You need to publish into a folder, zip that one and then command might work.
I personally switched to using [WAWSDeploy](https://github.com/davidebbo/WAWSDeploy) instead (it removes the need to zip the thing for a start 😉 ).
Cheers.
